packages/client/src/clients/guide/client.ts

```diff
@@ -76,18 +78,59 @@ const checkForWindow = () => {
 export const guidesApiRootPath = (userId: string | undefined | null) =>
   `/v1/users/${userId}/guides`;
 
-// Detect debug params like "knock_guide_key" from URL.
+// Detect debug params from URL or local storage
 const detectDebugParams = (): DebugState => {
   const win = checkForWindow();
   if (!win) {
     return { forcedGuideKey: null, previewSessionId: null };
   }
 
   const urlParams = new URLSearchParams(win.location.search);
-  const forcedGuideKey = urlParams.get(DEBUG_QUERY_PARAMS.GUIDE_KEY);
-  const previewSessionId = urlParams.get(DEBUG_QUERY_PARAMS.PREVIEW_SESSION_ID);
+  const urlGuideKey = urlParams.get(DEBUG_QUERY_PARAMS.GUIDE_KEY);
+  const urlPreviewSessionId = urlParams.get(
+    DEBUG_QUERY_PARAMS.PREVIEW_SESSION_ID,
+  );
+
+  // If URL params exist, persist them to localStorage and return them
+  if (urlGuideKey || urlPreviewSessionId) {
+    if (win.localStorage) {
+      try {
+        const debugState = {
+          forcedGuideKey: urlGuideKey,
+          previewSessionId: urlPreviewSessionId,
+        };
+        win.localStorage.setItem(DEBUG_STORAGE_KEY, JSON.stringify(debugState));
+      } catch {
+        // Silently fail in privacy mode
+      }
+    }
+    return {
+      forcedGuideKey: urlGuideKey,
+      previewSessionId: urlPreviewSessionId,
+    };
+  }
 
-  return { forcedGuideKey, previewSessionId };
+  // Check local storage if no URL params
+  let storedGuideKey = null;
+  let storedPreviewSessionId = null;
+
+  if (win.localStorage) {
+    try {
+      const storedDebugState = win.localStorage.getItem(DEBUG_STORAGE_KEY);
+      if (storedDebugState) {
+        const parsedDebugState = JSON.parse(storedDebugState);
```

Comment:
Could we create a helper like `safeJsonParse` that wraps this in try/catch, and return `undefined` in case of any parse error instead of throwing so we never crash from this?

Reply:
Yep, good suggestion, I added a method to make sure we always return `DebugState` after parsing